---
 src/nvcore/Debug.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nvcore/Debug.cpp b/src/nvcore/Debug.cpp
index 7379971..f96b937 100644
--- a/src/nvcore/Debug.cpp
+++ b/src/nvcore/Debug.cpp
@@ -665,6 +665,9 @@ namespace
 #  elif NV_CPU_PPC
         ucontext_t * ucp = (ucontext_t *)secret;
         return (void *) ucp->uc_mcontext.regs->nip;
+#    elif NV_CPU_ARM
+        ucontext_t * ucp = (ucontext_t *)secret;
+        return (void *) ucp->uc_mcontext.arm_pc;
 #    elif NV_CPU_AARCH64
         ucontext_t * ucp = (ucontext_t *)secret;
         return (void *) ucp->uc_mcontext.pc;
@@ -1021,7 +1024,7 @@ void debug::dumpCallstack(MessageHandler *messageHandler, int callstackLevelsToS
         writeStackTrace(trace, size, callstackLevelsToSkip + 1, lines);     // + 1 to skip the call to dumpCallstack
 
         for (uint i = 0; i < lines.count(); i++) {
-            messageHandler->log(lines[i], NULL);
+            va_list empty_va_list = {};
+            messageHandler->log(lines[i], empty_va_list);
             delete lines[i];
         }
     }
-- 
